### DISCUSSION TITLE
Who defines the syrup options?

### DISCUSSION THREAD

**Discussion by ConfusedMerlin** [DeviceFarmer/stf#615] [Q&A]
Greetings,

trying to track down devices, that do not have a connectionUrl (and a grey screen) despite being shown in the farm as connected and being reachable by adb container without problem.

So I made may way through the process flow, from the device list on the web gui, to the swagger get vor api/v1/devcies, to the rethinkDB access for the devices table. Now I am looking for the source of the connectURL, which has led me to a file ./lib/units/device/plugins/connect.js. 

It creates a plugin.start(), which gets called when a ConnectStartMessage is wired in. When that happens, a ConnectStartedMessage is wired out (which is received by a piece of code that inserts its url param into the db) as... positive Proimse then function, which gets the url from one of the two returns inside plugin.start().

So far, so good. But... the url to be send out with the ConnectStartedMessage is peeled out of the options parameter at the very top of the file, before its returned for the Promis (that bluebird stuff). This options parameter seems to come in with the syrup's instantiation. Now I stranded in the stf.syrup repository without a clue, about who actually fills the dependencies of the serial/parallel syrup class defined there. 

__Or to put it more blatant: who puts the the URL into the options parameter of stf.syrup which is used by the connect.js?__

Because... I suspect that part is responsible for not providing the current ConnectURL in the cases I described at the top of this question.

**Comment by denis99999** [comment]
@ConfusedMerlin ,
Please state clearly what your problem is by filling in the issue template? Anyway, if you are experiencing a gray screen, it means you have a communication problem between your device provider and your main machine, so start fixing it because the rest is just a side effect.

**Reply by ConfusedMerlin** [reply]
@denis99999 , this is the kind of problem where the issue would read like "cannot reproduce or even describe the circumstances it happens, because it happens appears random" . That is why I decided to search for its reason by digging through the code this time. 

And that is why - at the moment - I would like to learn where the syrup informations are coming from. Probably I will get stuck sooner or later again, but at least I might be able to learn more about the inner workings of the Devicefarm this way.

**Comment by ConfusedMerlin** [comment]
Okay, I now understand that this was the wrong question... we can ignore this question, as I must ask something different first.
